### PR TITLE
time-admin: shortcuts, images & shared code for buttons

### PR DESCRIPTION
**Pull request by rbuj**
- Add & enable mnemonic accelerator keys
- Show Save/Close button icons in Time/Date manager window
- Remove DialogAddButtonWithIconName from shares

### CODE REVIEW

File: capplets/time-admin/src/time-zone.c [hunk 1]
```diff
@@ -437,15 +437,21 @@ void SetupTimezoneDialog(TimeAdmin *ta)
...
-                                               GTK_RESPONSE_OK);
+
+    ta->TZclose = gtk_button_new_with_mnemonic (_("_Close"));
+    gtk_button_set_image (GTK_BUTTON (ta->TZclose), gtk_image_new_from_icon_name ("window-close", GTK_ICON_SIZE_BUTTON));
+    gtk_button_set_use_underline (GTK_BUTTON (ta->TZclose), TRUE);
+    gtk_style_context_add_class (gtk_widget_get_style_context (ta->TZclose), "text-button");
+    gtk_widget_set_can_default (ta->TZclose, TRUE);
+    gtk_dialog_add_action_widget (GTK_DIALOG (ta->dialog), ta->TZclose, GTK_RESPONSE_CANCEL);
+
+    ta->TZconfire = gtk_button_new_with_mnemonic (_("Con_firm"));
```
Comment (zhuyaliang) [inline capplets/time-admin/src/time-zone.c]: It should be ("_Confirm")

Comment (rbuj) [reply capplets/time-admin/src/time-zone.c]: You can't use the same the same key accelerator twice.

**Comment by zhuyaliang** [comment]
Why not encapsulate it as a function? Is there any benefit in doing so?

**Comment by rbuj** [comment]
You are right, we can see two blocks of code that they seem to be duplicated, as we build two buttons with similar code. It's done on form build, and the code encapsulation here can obfuscate the initialization of the controls. Do you want a private function/method to build the buttons?

**Comment by zhuyaliang** [comment]
This ```DialogAddButtonWithIconName``` function can fully meet the needs, why remove it?Encapsulated in the form of functions, the code has better readability.

**Comment by rbuj** [comment]
> This `DialogAddButtonWithIconName` function can fully meet the needs, why remove it?Encapsulated in the form of functions, the code has better readability.

In the same manner you didn't encapsulate buttons build in `static GtkWidget *SetDate(TimeAdmin *ta);`, and`DialogAddButtonWithIconName` should be local.

**Comment by zhuyaliang** [comment]
Thank you for adding icons to the buttons, but what I want to say is that a lot of duplicated code is not beautiful, and I think they can be encapsulated as a common function.

**Comment by rbuj** [comment]
Yes, the next natural step could be to write the ui files using glade.

**Comment by rbuj** [comment]
WIP time-admin.ui - Only for testing:
https://github.com/mate-desktop/mate-control-center/tree/test-time-admin-ui

**Pull request merged**